### ISSUE TITLE
[Export] tag is not working on object with script .(Cannot access a disposed object)

### ISSUE THREAD

**Issue by Mupli** [godotengine/godot#82438]
### Godot version

4.1.1

### System information

Linux Mint, C# , Forward +

### Issue description

**structure:**

Panel: **Window**  (some business logic - script)
- Control: content
- Panel: close (script (HideViewOnClick) + [Export] with attached **Window**)

When I click "close" I receive error. 

```
Object name: 'Godot.Panel'.
  <C# Error>     System.ObjectDisposedException
  <C# Source>    /root/godot/modules/mono/glue/GodotSharp/GodotSharp/Core/GodotObject.base.cs:73 @ IntPtr Godot.GodotObject.GetPtr(Godot.GodotObject )
  <Stack Trace>  GodotObject.base.cs:73 @ IntPtr Godot.GodotObject.GetPtr(Godot.GodotObject )
                 GodotObject.cs:422 @ UInt64 Godot.GodotObject.GetInstanceId()
                 **HideViewOnClick**.cs:20 @ void XXX.**HideViewOnClick**._GuiInput(Godot.InputEvent)
                 
```                 
```csharp
public partial class HideViewOnClick : Control
{
    [Export]
    private Control view;


    public override void _GuiInput(InputEvent @event)
    {
        base._GuiInput(@event);
        if (@event is InputEventMouseButton)
        {
            var e = @event as InputEventMouseButton;
            if (e.IsReleased() && e.ButtonIndex == MouseButton.Left)
            {
                view.Visible = false;
            }
        }
    }

}
```

### Steps to reproduce

1. create scene as in the issue description
2. try to hide window

Optionally try to add scene as "Prefab/Template" 

### Minimal reproduction project

N/A

**Comment by Mupli** [comment]
Can be closed resolved issue with 

```csharp
[Export] NodePath  path;
..
var view = getNode<Control>(path);

```

**Comment by Thomas-X** [comment]
Tried to reproduce, seems it might  already be fixed in 4.1.2. 
![image](https://github.com/godotengine/godot/assets/21952645/c4394a4d-c189-4784-aa2a-98c55d0438ae)

**Comment by Mupli** [comment]
> Tried to reproduce, seems it might already be fixed in >=4.1.2. ![image](https://user-images.githubusercontent.com/21952645/277327105-c4394a4d-c189-4784-aa2a-98c55d0438ae.png)

No ...it is not working.Though I have different error but intent is the same. Change the private "Control" to "Button". Something that is not casting directly. 


So: 

Example:
b1 (Button) has attached script A (valid)

```C#
public class A: Control {
}
``` 

Other file...
```C#
[Export] 
Button b1; // error.  Cannot cast object
...
b1.Text = "Some button text";
```

Oh well it can be resolved with : /*Control*/ control.Set("text", "some text")

**Comment by Mupli** [comment]
Can be closed as my first issue is resolved. 

Second issue  is not resolvable in standard C# or it will take lives of countless developers :D.
